vm.go

```diff
@@ -433,8 +463,40 @@ func (session *ioSession) WaitForShim() error {
 	select {
 	case <-session.shimConnected:
 	case <-time.After(waitForShimTimeout):
-		return fmt.Errorf("timeout waiting for shim with token %s", session.token)
+		msg := fmt.Sprintf("timeout waiting for shim with token %s", session.token)
+		session.vm.infof(1, "session", msg)
+		// No need to call session.ResetShim() here. We time out because we haven't
+		// seen a shim. This error will be reported to the runtime.
+		return errors.New(msg)
 	}
+
+	return nil
+}
+
+var waitForExecTimeout = 30 * time.Second
+
+// WaitForProcess will wait until the process inside the VM is fully started. If
+// it's already the case, WaitForProcess will return immediately.
+// shouldReset controls if we should reset the internal shim state on timeout.
+// It should be set to true in code path that will close the shim connection on
+// timeout.
+func (session *ioSession) WaitForProcess(shouldReset bool) error {
+	session.vm.infof(1, "session",
+		"waiting for runtime to execute the process for token %s (timeout %s)",
```

Comment:
Is that `%s` correct? Even if it is, it might look less odd if the format was specified as `%v`.